packages/playwright/src/runner/vcs.ts

```diff
@@ -27,6 +27,20 @@ export async function detectChangedTestFiles(baseCommit: string, configDir: stri
       ).split('\n').filter(Boolean);
     } catch (_error) {
       const error = _error as childProcess.SpawnSyncReturns<string>;
+
+      const unknownRevision = error.output.some(line => line?.includes('unknown revision'));
+      if (unknownRevision) {
+        const isShallowClone = childProcess.execSync('git rev-parse --is-shallow-repository', { encoding: 'utf-8',  stdio: 'pipe' }).trim() === 'true';
+        if (isShallowClone) {
+          throw new Error([
+            `Revision '${baseCommit}' is not available in the local repository.`,
+            `On CI, this is likely caused by a shallow clone.`,
+            `To fix, clone the full repository history:`,
+            '   https://github.com/actions/checkout#fetch-all-history-for-all-tags-and-branches'
```

Comment:
```suggestion
            'e.g. for GitHub Actions: https://github.com/actions/checkout#fetch-all-history-for-all-tags-and-branches'
```

maybe something like that. Let's also get a stamp from Dima for this.